Codespace setup

jeroenminderman opened the pull request:
## Review comments

### Description of changes -

- This PR combines and supercedes #2 
- Updated requirements.txt to deal with version issues between packages
- Added nltk.downloads of corpora to setup.py to ensure this works out of the box
- Added working Codespace setup (devcontainer config) that allows automatic set up of fully working environment, to the point user can spin up the Codespace and run coder.py immediately.

### Checklist

- [x] **Documentation**: code includes docstrings and examples.
- [x] **Installation**: Installation or build of the code succeeds.
- [x] **Functionality**: Any functional claims of new code have been confirmed.
- [x] **Automated tests**: Unit tests cover essential functions for a reasonable range
  of inputs and conditions. All tests pass on your local machine and/or cloud setup/Codespace.
- [x] **Packaging guidelines**: New code conforms to the project contribution
  guidelines.

josh-ons2022 commented:
@jeroenminderman only something small but would be good to add to the readme the code to activate the venv

jeroenminderman commented:
> @jeroenminderman only something small but would be good to add to the readme the code to activate the venv
> 
> * Able to run through to processed.csv successfully with creation of venv with no issues
> * In relation to system_setup.sh is there documentation for how to do this?

Thanks @josh-ons2022 -
`system_setup.sh` isn't intended to be run by a user direct but is purely used in the automated standup of a new Codespace instance. But noted on the documentation - I'll add a bit on how to run the Codespace in the first place.

josh-ons2022 commented:
> @josh-ons2022 I've made some additions to the README, could you have a quick look to see if you're happy please, and if yes, can you submit/approve your review?

Sure will do

jeroenminderman commented:
Description of changes -

- This PR combines and supercedes https://github.com/datasciencecampus/occupationcoder-international/pull/2
- Updated requirements.txt to deal with version issues between packages
- Added nltk.downloads of corpora to setup.py to ensure this works out of the box
- Added working Codespace setup (devcontainer config) that allows automatic set up of fully working environment, to the point user can spin up the Codespace and run coder.py immediately.
- Added doc notes for Codespace
